### DISCUSSION TITLE
Cable Trace Question

### DISCUSSION THREAD

**Discussion by theredon** [netbox-community/netbox#15450] [Q&A] [answered]
Is there a way to add a custom field to display in the cable trace? 

![image](https://github.com/netbox-community/netbox/assets/24547976/da77077a-1775-49db-a7c9-470a78073068)

We use a taping system to identify tower cables and I've built a custom field to add it to the cables, but would like it to show up here as well if possible.  I haven't been able to find any way to do it, so I'm guessing my best bet would be to create a report to export the details to a CSV instead of being about to download the SVG with the information on it, but figured I'd ask just in case.

**Comment by candlerb** [comment, marked as answer]
No, it's not templated. For example, in order to get circuit descriptions added I needed to raise a feature request: #12831

Here's the code which labels a cable, in `netbox/dcim/svg/cables.py`:

```
        # If the cable count is higher than 2, collapse the description into a tooltip
        if cable_count > 2:
            ... snip
        else:
            labels = [
                f'Cable {cable}',
                cable.get_status_display()
            ]
            description = []
            if cable.type:
                labels.append(cable.get_type_display())
            if cable.length is not None and cable.length_unit:
                # Include the cable length in the tooltip
                labels.append(f'{cable.length} {cable.get_length_unit_display()}')
```

and in turn, `{cable}` invokes `Cable.__str__` from `netbox/dcim/models/cables.py`

```
    def __str__(self):
        pk = self.pk or self._pk
        return self.label or f'#{pk}'
```

In the absence of a hook to format the SVG labels, and without wanting to maintain your own fork of Netbox, you're probably best duplicating the logic in a plugin (or as you say, export and then render externally)
